strongbox-security/strongbox-user-management/src/main/java/org/carlspring/strongbox/authorization/AuthorizationConfigFileManager.java

```diff
@@ -45,4 +55,58 @@ public ConfigurationResourceResolver getConfigurationResourceResolver()
         return configurationResourceResolver;
     }
 
+    @Override
+    public synchronized void store(AuthorizationConfigDto configuration) throws IOException
+    {
+        Map<SystemRole, String> changed = getChangedRestrictedRoles(configuration);
+        if (changed.size() > 0)
+        {
+            String message = String.join("; ", changed.values());
+            throw new RuntimeException(message);
+        }
+        super.store(configuration);
+    }
+
+    public Map<SystemRole, String> getChangedRestrictedRoles(AuthorizationConfigDto configuration) throws IOException {
```

Comment:
there should be enough to return just `Set<SystemRole>` here without custom messages as map values (see above).

Reply:
also I suggest to make this method `private`.

Reply:
Set - done. Private is hard to test, best that i came up with is @VisibleForTesting annotation. Any suggestions?

Reply:
I suggest to use `AuthorizationConfigFileManager.store` method, the test cases would be:
1. Success on regular role updated
2. Success on regular role deleted
3. Fail on system role updated
4. Fail on system role deleted

Reply:
@sbespalov I don't khow how to test this method, because of super.store() call. I can move super.store() call to the separate method and stub it with Mockito.spy. Is there any better ways?

Reply:
@BogdanSukonnov I suggest to move `ConfigurationResourceResolver` dependency into constructor and use mocked `Resource` instead.